starport-core/src/main/scala/com/krux/starport/db/tool/SubmitPipeline.scala

```diff
@@ -2,50 +2,68 @@ package com.krux.starport.db.tool
 
 import java.io.File
 import java.net.URLClassLoader
+import java.security.Permission
 
 import com.github.nscala_time.time.Imports._
-import scopt.OptionParser
 import slick.jdbc.PostgresProfile.api._
-
-import com.krux.hyperion.cli.Reads._
-import com.krux.hyperion.expression.Duration
 import com.krux.hyperion.{DataPipelineDefGroup, RecurringSchedule, Schedule}
+import com.krux.starport.{ErrorExit, Logging}
 import com.krux.starport.config.StarportSettings
 import com.krux.starport.db.record.Pipeline
 import com.krux.starport.db.table.Pipelines
 import com.krux.starport.db.{DateTimeMapped, WaitForIt}
-import com.krux.starport.util.notification.SendSlackMessage
 import com.krux.starport.util.{DateTimeFunctions, S3FileHandler}
-import com.krux.starport.{BuildInfo, ErrorExit, Logging}
+import com.krux.starport.util.notification.SendSlackMessage
 
+sealed case class ExitException(status: Int) extends Exception("[substitute for sys.exit]") {}
+
+/**
+  * Override SecurityManager when executing via AWS Lambda, in order to "catch" sys.exit.
+  * Note that in the SubmitPipelineOptionParser * when the --help flag is called execution
+  * flow will still allow the processing of other flags.
+  *
+  * If -Dexecution.context=lambda property must is set, main will enable this class
+  */
+sealed class NoExitSecurityManager extends SecurityManager {
+  override def checkPermission(perm: Permission): Unit = {}
+  override def checkPermission(perm: Permission, context: Object): Unit = {}
+  override def checkExit(status: Int): Unit = {
+    throw ExitException(status)
+  }
+}
 
 object SubmitPipeline extends DateTimeFunctions with WaitForIt with DateTimeMapped with Logging {
 
-  private val ValidEmail = """^([a-zA-Z0-9.!#$%&’'*+/=?^_`{|}~-]+)@([a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*)$""".r
-
   lazy val starportSettings = StarportSettings()
 
-  case class Options(
-    jar: String = "",
-    pipelineObject: String = "",
-    startNow: Boolean = false,
-    update: Boolean = false,
-    noBackfill: Boolean = false,
-    baseDir: Option[String] = None,
-    cleanUp: Boolean = false,
-    schedule: Option[DateTime] = None,
-    frequency: Option[Duration] = None,
-    enable: Option[Boolean] = None,
-    owner: Option[String] = None,
-    dryRun: Boolean = false
-  ) {
-    def backfill = !noBackfill
+  def main(args: Array[String]): Unit = {
+    val executionContext = System.getProperty("execution.context")
+    logger.debug(s"args: ${args.mkString(",")}")
+    logger.debug(s"context: ${executionContext}")
+    if (executionContext == "lambda") {
+      System.setSecurityManager(new NoExitSecurityManager)
+    }
+    SubmitPipelineOptionParser.parse(args, errorHandler) match {
+      case Some(options) => {
+        run(options)
+      }
+      case None =>
```

Comment:
If there's nothing to be done for the `None` case, you can simply do `SubmitPipelineOptionParser.parse(args, errorHandler).foreach(run)`

Reply:
cleaned up